### PR TITLE
Aws task6 sns sqs

### PR DESCRIPTION
**Pull request by colpachoque**
**Task 6**
- [x] File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [x] File serverless.yml contains configuration for SQS catalogItemsQueue
- [x] File serverless.yml contains configuration for catalogBatchProcess function
- [x] File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription

[Link to the FE app](https://dqx938xovkpeh.cloudfront.net/admin/products)

### CODE REVIEW

**Comment by kotmatpockuh** [comment]
4. is it correct?

File: another-import-service/import-file-parser.js [hunk 1]
```diff
@@ -31,7 +31,23 @@ export const importSecondFileParser = async event => {
...
 
-        const data = await collectDataForLogging(record.s3.object.key);
+        const data = await collectDataForLogging(record.s3.object.key),
+            sqs = new AWS.SQS();
+
+
+        for (const product of data) {
+            try {
+                const res = await sqs.sendMessage({
+                    QueueUrl: 'https://sqs.eu-west-1.amazonaws.com/517465834650/catalogItemsQueue',
```
Comment (kotmatpockuh) [inline another-import-service/import-file-parser.js:41]: This should not be hardcoded! And should be received from CloudFormation automatically ;) with a correct serverless config ofc.

File: another-import-service/serverless.yml [hunk 2]
```diff
@@ -25,6 +25,11 @@ provider:
         - 's3:*'
       Resource: 
         - 'arn:aws:s3:::colpachoques-bucket-imported/*'
+    - Effect: 'Allow'
+      Action:
+        - 'sqs:*'
+      Resource:
+        - 'arn:aws:sqs:eu-west-1:517465834650:catalogItemsQueue'
```
Comment (kotmatpockuh) [inline another-import-service/serverless.yml:32]: Should be dynamic 🤷‍♂️

File: product-service/catalog-batch-process.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		client = new Client(dbOptions),
+		sns = new AWS.SNS({ region: 'eu-west-1'})
+
+	console.log(products);
+
+	try {
+		await client.connect();
+		await client.query('BEGIN');
+
+		for (const product of products) {
```
Comment (kotmatpockuh) [inline product-service/catalog-batch-process.js:28]: I guess this should be refactored ;) as a single create is almost the same :)

File: product-service/package.json [hunk 4]
```diff
@@ -3,6 +3,7 @@
   "version": "1.0.0",
   "description": "Serverless product service",
   "dependencies": {
+    "aws-sdk": "^2.797.0",
```
Comment (kotmatpockuh) [inline product-service/package.json:6]: Please move it to devDeps to reduce bundle size.

File: product-service/serverless.yml [hunk 5]
```diff
@@ -36,6 +51,28 @@ custom:
...
+      Properties:
+        QueueName: catalogItemsQueue
+    SNSTopic:
+      Type: AWS::SNS::Topic
+      Properties:
+        TopicName: createProductTopic
+    SNSSubscription:
+      Type: AWS::SNS::Subscription
+      Properties:
+        Endpoint: krakoryak@gmail.com
```
Comment (kotmatpockuh) [inline product-service/serverless.yml:67]: Move to .env 😛

File: another-import-service/import-file-parser.js (hunk 1 above)
Comment (kotmatpockuh) [reply another-import-service/import-file-parser.js:41]: Example: 

```
- Effect: Allow
          Action:
              - 'sns:*'
          Resource:
              - Fn::ImportValue: SNSTopic
```

where SNSTopic ->

```
SNSTopic:
            Type: AWS::SNS::Topic
            Properties:
                TopicName: createProductTopic
```

and 

```
environment:
        SQS_URL:
            Ref: SQSQueue
```

File: product-service/serverless.yml [hunk 6]
```diff
@@ -25,7 +25,22 @@ provider:
...
+        - 'sqs:*'
+      Resource:
+        - Fn::GetAtt: [catalogItemsQueue, Arn]
+    - Effect: 'Allow'
+      Action:
+        - 'sns:*'
+      Resource:
+        Ref: SNSTopic
+  environment:
+    SQS_URL:
```
Comment (skyRoma) [inline product-service/serverless.yml:40]: I think it's not needed

File: another-import-service/import-file-parser.js (hunk 1 above)
Comment (skyRoma) [reply another-import-service/import-file-parser.js:41]: or like this: 
```
   environment: {
      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
      SQS_QUEUE_URL: {
        'Fn::ImportValue': 'SQSUrl',
      },
    },
```

Where `SQSUrl` is your output property from another serverless file

**Review by skyRoma — APPROVED** [review]

**Pull request merged**